libwasmvm/src/cache.rs

```diff
@@ -693,18 +699,17 @@ mod tests {
         let mut error_msg = UnmanagedVector::default();
         let metrics = get_metrics(cache_ptr, Some(&mut error_msg));
         let _ = error_msg.consume();
-        assert_eq!(
-            metrics,
-            Metrics {
-                hits_pinned_memory_cache: 0,
-                hits_memory_cache: 0,
-                hits_fs_cache: 1,
-                misses: 0,
-                elements_pinned_memory_cache: 1,
-                elements_memory_cache: 0,
-                size_pinned_memory_cache: 5665691,
-                size_memory_cache: 0,
-            }
```

Comment:
@uint are you aware of an implementation similar to this `assert_approx_eq` but with a relative epsilon? I don't want to say 5665691 +/- 665691 but 5665691 +/- 15%. We could use that for some gas tests as well.

Reply:
Jakub or Bart (I don't remember) actually did that for Isotonic. It's easy to implement. I'd just write our own macro in `cosmwasm` and maybe open a PR to the `assert_approx_eq` crate.

Reply:
That would be much appreciated!